lib/tts_isolate.dart

```diff
@@ -62,25 +63,44 @@ void onStart(ServiceInstance service) async {
     prefs.getString('tts_channel', defaultValue: '{}').listen((channel) async {
       debugPrint("tts_channel changed to: $channel");
       if (channel.isNotEmpty && channel != "{}") {
-        // Fetch messages from Firestore
-        // var messages = await fetchMessagesFromFirestore(channel);
-        // print('MESSAGES HERE $messages');
-        // Process messages as needed
+        listenForNewMessages(channel);
       }
     });
   });
 }
 
-Future<List<dynamic>> fetchMessagesFromFirestore(String channelId) async {
-  List<dynamic> messages = [];
+void listenForNewMessages(String channelId) {
+  // Listen to changes in Firestore and process messages
+  FirebaseFirestore.instance
+      .collection('channels')
+      .where('channelId', isEqualTo: channelId)
+      .snapshots()
+      .listen((QuerySnapshot<Map<String, dynamic>> snapshot) {
```

Comment:
This is almost correct, nice work. The code as written now won't unsubscribe when the channel changes though. You can fix this with `.switchMap()` from the `stream_transforms` package in the caller:

```
import 'package:stream_transforms/stream_transforms.dart';

prefs.getString('tts_channel', default_value: '{}')
  .switchMap((channel) async {
    return FirebaseFirestore.instance
      .collection('channels')
      .where('channelId', isEqualTo: channel.id)
      .snapshots();
    })
  .listen((snapshot) {
    ...
  });
```

Here is the documentation for `switchMap()`: https://pub.dev/documentation/stream_transform/latest/stream_transform/Switch/switchMap.html

And here's a more insightful explanation: https://stackoverflow.com/questions/61569917/how-do-i-nest-streams-in-dart-map-streams-to-stream-events